libcloud/dns/drivers/godaddy.py

```diff
@@ -0,0 +1,521 @@
+        zone = Zone(id=item['domainId'], domain=item['domain'],
+                    type='master', ttl=None,
+                    driver=self, extra=extra)
+        return zone
+
+    def _to_records(self, items, zone=None):
+        records = []
+
+        for item in items:
+            records.append(self._to_record(item=item, zone=zone))
+        return records
+
+    def _to_record(self, item, zone=None):
+        ttl = item['ttl']
+        type = self._string_to_record_type(item['type'])
+        name = item['name']
+        id = '{0}:{1}'.format(name, type)
+        record = Record(id=id, name=name,
+                        type=type, data=item['data'],
+                        zone=zone, driver=self,
+                        extra={'ttl': ttl})
+        return record
+
+    def _to_tlds(self, items):
+        tlds = []
+        for item in items:
+            tlds.append(self._to_tld(item))
+        return tlds
+
+    def _to_tld(self, item):
+        return GoDaddyTLD(
+            name=item['name'],
+            tld_type=item['type']
+        )
+
+    def list_zones(self):
+        """
+        Return a list of zones (purchased domains)
+
+        :return: ``list`` of :class:`Zone`
+        """
+        result = self.connection.request(
+            '/v1/domains/').object
+        zones = self._to_zones(result)
+        return zones
+
+    def list_records(self, zone):
+        """
+        Return a list of records for the provided zone.
+
+        :param zone: Zone to list records for.
+        :type zone: :class:`Zone`
+
+        :return: ``list`` of :class:`Record`
+        """
+        result = self.connection.request(
+            '/v1/domains/%s/records' % zone.domain).object
+        records = self._to_records(items=result, zone=zone)
+        return records
+
+    def create_record(self, name, zone, type, data, extra=None):
+        """
+        Create a new record.
+
+        :param name: Record name without the domain name (e.g. www).
+                     Note: If you want to create a record for a base domain
+                     name, you should specify empty string ('') for this
+                     argument.
+        :type  name: ``str``
+
+        :param zone: Zone where the requested record is created.
+        :type  zone: :class:`Zone`
+
+        :param type: DNS record type (A, AAAA, ...).
+        :type  type: :class:`RecordType`
+
+        :param data: Data for the record (depends on the record type).
+        :type  data: ``str``
+
+        :param extra: Extra attributes (driver specific). (optional)
+        :type extra: ``dict``
+
+        :rtype: :class:`Record`
+        """
+        if extra is None:
+            extra = {}
+        new_record = {}
+        if type is RecordType.SRV:
+            new_record = {
+                'type': type,
+                'name': name,
+                'data': data,
+                'priority': 1,
+                'ttl': extra['ttl'] if hasattr(extra, 'ttl') else 5,
```

Comment:
I just noticed you use `hasattr`. This won't work since `hasattr` only works on class instances. For dictionaries you should either use `key in dict` or `dict.get(key, default_value)`.